app/models/task.py

```diff
@@ -3,4 +3,31 @@
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable=True)
+    
+
+    def to_json(self):
+       
+        if self.completed_at == None:
+            is_complete = False
+        else:
+            is_complete = True 
+        if self.goal_id:
+            return {
+                "id": self.task_id,
+                "goal_id": self.goal_id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": is_complete
+            }
+        return {
+            "id": self.task_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": is_complete
+        }
```

Comment:
Notice there is repeated code here. Consider refactoring by creating a variable to hold the dictionary such as 
```
json = 
{"id": self.task_id,           
 "title": self.title,            
"description": self.description,           
"is_complete": is_complete  }
```
and then conditionally adding the goal info 
```
if self.goal_id:
    json["goal_id"] = self.goal_id
```

and then  ```return json```